apt/bagit_archive.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import tempfile
+import time
+import zipfile
+from pathlib import Path
+from typing import Any, ClassVar
+
+import bagit  # type: ignore[import]
+
+from apt.config import Config
+from apt.utils import stream_file_transfer
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Config()
+
+
+class BagitArchive:
+    """Class for creating a Bagit archive files."""
+
+    DEFAULT_CHECKSUMS: ClassVar[list[str]] = [
+        "md5",
+        "sha256",
+    ]
+
+    def __init__(
+        self,
+        bag_metadata: dict | None = None,
+    ):
+        self.bag_metadata = bag_metadata or {"Contact-Name": "Default Contact"}
+
+        self.bag = None
+        self.bag_path: Path | None = None
+
+    def download_file(self, source_uri: str, target_path: str | Path) -> Path:
```

Comment:
Related to another comment: In using `target_path` instead of `target_uri`, does this mean `BagitArchive.download_file` should only download files to "filesystem path" (i.e., it shouldn't be used to "download a file" to S3)? 🤔 Since `BagitArchive.download_file` uses `stream_file_transfer`, seems like it could...

Reply:
I like the renames in https://github.com/MITLibraries/archival-packaging-tool/pull/10/commits/3e3d38cffa677e6ba11a5d0f59a0a0d178db25f2! Thank you for these updates!